src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/InputInProcess.cs

```diff
@@ -102,11 +90,17 @@ internal async Task SendToNavigateToAsync(Action<IInputSimulator> callback)
             TestServices.JoinableTaskFactory.Run(async () =>
             {
                 await TestServices.JoinableTaskFactory.SwitchToMainThreadAsync();
-                var searchBox = Assert.IsAssignableFrom<TextBox>(Keyboard.FocusedElement);
-                Assert.Equal("PART_SearchBox", searchBox.Name);
+                var searchBox = Assert.IsAssignableFrom<Control>(Keyboard.FocusedElement);
+                Assert.Contains(searchBox.Name, new[] { "PART_SearchBox", "SearchBoxControl" });
             });
 
-            callback(new InputSimulator());
+            var inputSimulator = new InputSimulator();
+            foreach (var key in keys)
+            {
+                key.Apply(inputSimulator);
+
+                await Task.Delay(1000);
```

Comment:
Is there any problem if the input is too fast?

Reply:
I'll add a comment. The AllInOneSearch updates the results asynchronously, so without the delay the Return key will be processed before the desired search item is in the results.

Reply:
Can we instead have something stronger that says if the key being sent is a return, we actually wait until the results have settled down?